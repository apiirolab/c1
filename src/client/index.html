<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>