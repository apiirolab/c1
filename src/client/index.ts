export const info = 'Client Info';
