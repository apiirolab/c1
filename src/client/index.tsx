import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
