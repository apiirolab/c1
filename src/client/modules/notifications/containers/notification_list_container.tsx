import * as NOTIFICATIONS from 'data/client/notifications.schema.graphql';
import * as React from 'react';

import { Query } from 'react-apollo';

import { Notification } from '../../../../data/generated/yoga';
import { NotificationListView } from '../views/notification_list_view';

export const QUERY = NOTIFICATIONS;

type Data = {
  notifications: Notification[]
}
class NotificationsQuery extends Query<Data> {}

export const NotificationsContainer: React.SFC = () => (
  <NotificationsQuery query={NOTIFICATIONS} variables={{ start:0, end: 10}}>
    {({ loading, error, data }) => {
      if (loading) { return 'Loading...'; }
      if (error) { return `Error! ${error.message}`; }

      return (
        <NotificationListView notifications={data.notifications} />
      );
    }}
  </NotificationsQuery>
);