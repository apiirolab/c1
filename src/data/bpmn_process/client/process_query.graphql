query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
