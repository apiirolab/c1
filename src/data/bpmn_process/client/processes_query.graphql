query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    id
    name
  }
}
