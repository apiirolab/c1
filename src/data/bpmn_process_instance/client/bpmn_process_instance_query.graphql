fragment BpmnProcessResource on Resource {
  name
  type
  id
  content
}

fragment BpmnProcessComment on Comment {
  id
  date
  replyTo
  text
  user {
    name
  }
}

fragment BpmnProcessTask on BpmnTaskInstance {
  name
  dateStarted
  dateFinished
  performer {
    name
  }
  performerRoles
}

fragment BpmnProcessDefinition on BpmnProcess {
  name
  description
  type
  model
  resources {
    ...BpmnProcessResource
  }
}

query BpmnProcessInstanceQuery($id: String!) {
  bpmnProcessInstance(id: $id) {
    id
    status
    process {
      ...BpmnProcessDefinition
    }
    owner {
      name
    }
    tasks {
      ...BpmnProcessTask
    }
    comments {
      ...BpmnProcessComment
    }
    resources
    dateStarted
    dateFinished
  }
}
