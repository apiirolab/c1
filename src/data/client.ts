import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
