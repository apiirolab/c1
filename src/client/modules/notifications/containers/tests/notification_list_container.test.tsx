// dog.test.js
import * as React from 'react';
import { create, mock, MockedProvider, Yoga } from 'tests/client';
import { NotificationsContainer, QUERY } from '../notification_list_container';

// import wait from 'waait';

describe('Notifications', () => {
  storyOf(
    'Container',
    {
      componentWithData(notifications: Yoga.Notification[] = null) {
        if (!notifications) {
          mock.expect(QUERY).reply({
            notifications: [{ id: '1', name: 'one' }, { id: '2', name: 'two' }]
          });
        }

        return (
          <MockedProvider>
            <NotificationsContainer />
          </MockedProvider>
        );
      }
    },
    data => {
      // xit('renders loading', () => {
      //   const container = create(
      //     <MockedProvider mocks={mocks()} addTypename={false}>
      //       <NotificationsContainer />
      //     </MockedProvider>
      //   );

      //   expect(container).toMatchSnapshot();
      // });

      it('renders final', async () => {
        // console.log(typeDefs);
        mock.expect(QUERY).reply({
          // <- `query` from the code above
          notifications: [{ id: '1', name: 'one' }, { id: '2', name: 'two' }]
        });

        const container = create(data.componentWithData());

        expect(container).toMatchSnapshot();
      });
    }
  );
});
