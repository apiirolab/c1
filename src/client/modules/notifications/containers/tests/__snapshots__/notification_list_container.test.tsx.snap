// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Container renders final 1`] = `
<div
  className="ui segment"
>
  <h5
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="bell icon"
    />
    <div
      className="content"
    >
      Notifications
    </div>
  </h5>
  <div
    className="ui divided list sc-bwzfXH flOgBY"
    role="list"
  >
    <div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <span
          className="sc-bdVaJa cWCCxy"
        >
          a few seconds ago
        </span>
         · 
      </div>
    </div>
    <div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <span
          className="sc-bdVaJa cWCCxy"
        >
          a few seconds ago
        </span>
         · 
      </div>
    </div>
  </div>
</div>
`;
